### ISSUE TITLE
Missing expression and invalid chars in error message

### ISSUE THREAD

**Issue by francisdb** [jmrozanec/cron-utils#109]
Tried to parse `* * -1 * * ?` and got the following message

```
java.lang.IllegalArgumentException: Invalid chars in expression! Expression:  Invalid chars:  (FieldParser.java:247)
[error] com.cronutils.parser.FieldParser.stringToInt(FieldParser.java:247)
[error] com.cronutils.parser.FieldParser.map(FieldParser.java:228)
[error] com.cronutils.parser.FieldParser.parseBetween(FieldParser.java:142)
[error] com.cronutils.parser.FieldParser.dashSplitResult(FieldParser.java:90)
[error] com.cronutils.parser.FieldParser.parse(FieldParser.java:83)
[error] com.cronutils.parser.CronParserField.parse(CronParserField.java:61)
[error] com.cronutils.parser.CronParser.parse(CronParser.java:95)
```

**Comment by espiegel** [comment]
Is -1 a valid cron value?

Are you surprised that you got an exception? Or are you referring to the exception message which doesn't list the "Invalid chars" ?

**Comment by francisdb** [comment]
See ticket title, the error is expected but the message is incomplete

**Comment by jmrozanec** [comment]
@francisdb Thank you for reporting the issue!
@espiegel thank you again for tackling migration jodatime->jdk8 time. Getting some issues after a big refactor is normal - please don't take it personal. If they arose, we just look at the root problem and fix them whenever possible. If a test did not cover this in a previous version, there is always room for improvement :) I am amazed how for each release @francisdb provides us quick feedback to keep improving.
If someone gets time to provide a fix soon, is more than welcome!

**Comment by francisdb** [comment]
hehe, most of the time it's switch version, log errors, rollback and wait for next minor version.
Once there are even more tests I guess this should get better.

**Comment by jmrozanec** [comment]
@francisdb @espiegel Just checked this issue and provided a workaround. We detect the issue and throw an exception, with the following message: "Failed to parse '* * -1 * * ?'. Invalid expression! Expression: -1 does not describe a range. Negative numbers are not allowed."  Thank you for reporting it.

**Comment by jmrozanec** [comment]
@francisdb @espiegel The fix was released with version 5.0.5
